docker/setup-ubuntu.sh

```
@@ -1,11 +1,92 @@
+fi
+
+echo "🔍 Detecting NVIDIA driver version..."
+DRIVER_VERSION=$(nvidia-smi --query-gpu=driver_version --format=csv,noheader,nounits | head -1)
+echo "Found NVIDIA driver: $DRIVER_VERSION"
+
+echo "🔍 Determine compatible CUDA version based on driver..." 
+get_compatible_cuda_version() {
+    local driver_ver=$1
+    
+    # Extract major version (e.g., 560 from 560.35.03)
+    local major_ver=$(echo $driver_ver | cut -d'.' -f1)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)